### DISCUSSION TITLE
Questions about Copilot + Open Source Software Hierarchy

### DISCUSSION THREAD

**Discussion by liaochris** [python-pillow/Pillow#7903] [General]
Hi! My name is Chris and I'm writing a thesis on Open Source Software. I'm trying to collect/validate my data and I have two questions for the maintainers of this project.
1) Did this project receive free github copilot access on June 2022?
2) My thesis is especially focused on understanding hierarchical structures. Would it be possible to share a list of individuals in this project with triage/write/maintain/admin access in this project? I understand this may be confidential information, so please feel free to share it with [chrisliao@uchicago.edu ](chrisliao@uchicago.edu)

Happy to chat further if you have questions, and thank you for your time!

**Comment by aclark4life** [comment]
> Hi! My name is Chris and I'm writing a thesis on Open Source Software. I'm trying to collect/validate my data and I have two questions for the maintainers of this project.
> 
>     1. Did this project receive free github copilot access on June 2022?


Not that I'm aware of


> 
>     2. My thesis is especially focused on understanding hierarchical structures. Would it be possible to share a list of individuals in this project with triage/write/maintain/admin access in this project? I understand this may be confidential information, so please feel free to share it with [chrisliao@uchicago.edu ](chrisliao@uchicago.edu)


Other than what is already public information, probably not. Further I'm not sure you actually need help from the maintainers to find that information. This is already a public repository.

> 
> 
> Happy to chat further if you have questions, and thank you for your time!

**Reply by liaochris** [reply]
That's interesting to know, thanks for sharing! Out of curiosity, what do you see when you check your current copilot eligibility ([https://github.com/github-copilot/free_signup](https://github.com/github-copilot/free_signup))?

**Reply by radarhere** [reply]
When I go to https://github.com/github-copilot/signup, I don't have to pick Pillow first, so the answer isn't specific to the project, it's just about my account.

**Reply by liaochris** [reply]
Yes, copilot access is account-specific, but only accounts that are maintainers of popular open source projects are eligible. Presumably, if the only top open source project (think thousands of stars - unfortunately, the exact criteria is what I'm trying to figure out) you're a maintainer of is Pillow, then your 1-year copilot license is implicitly tied to your status as a maintainer of Pillow. 
If you're a maintainer of other top open source projects, then it might not be tied to Pillow. Thanks for all the discussion - it's really helpful.

**Reply by wiredfool** [reply]
Looks like the python-pillow org is eligible to buy business access to copilot, for what that's worth. 

<img width="983" alt="image" src="https://github.com/python-pillow/Pillow/assets/19792/2281e4e6-6586-4577-b320-91320ab4cf94">
